pkgs/by-name/fl/flashprog/package.nix

```diff
@@ -24,19 +26,29 @@ stdenv.mkDerivation rec {
     installShellFiles
     pkg-config
   ];
-
   buildInputs = [
     libftdi1
-    libjaylink
     libusb1
   ] ++ lib.optionals (!stdenv.isDarwin) [
-    libgpiod_1
     pciutils
+  ] ++ lib.optionals with_jlink [
+    libjaylink
+  ] ++ lib.optionals (stdenv.isLinux && with_gpio) [
+    libgpiod
   ];
 
-  makeFlags = [ "PREFIX=$(out)" "libinstall" ]
-    ++ lib.optionals stdenv.isDarwin [ "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no" ]
-    ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ "CONFIG_INTERNAL_X86=no" "CONFIG_INTERNAL_DMI=no" "CONFIG_RAYER_SPI=0" ];
+  makeFlags = let
+    yesNo = flag: if flag then "yes" else "no";
+  in [
+    "libinstall"
+    "PREFIX=$(out)"
+    "CONFIG_JLINK_SPI=${yesNo with_jlink}"
+    "CONFIG_LINUX_GPIO_SPI=${yesNo (stdenv.isLinux && with_gpio)}"
```

Comment:
```suggestion
    "CONFIG_LINUX_GPIO_SPI=${yesNo with_gpio}"
```

Reply:
Same thing here. libgpio is specific to Linux and the condition needs to stay.